jaeger-core/src/main/java/com/uber/jaeger/reporters/RemoteReporter.java

```diff
@@ -187,4 +189,56 @@ public void close() {
       open = false;
     }
   }
+
+  public static class Builder implements Reporter.Builder {
+    private static final int DEFAULT_FLUSH_INTERVAL_MS = 1000;
+    private static final int DEFAULT_MAX_QUEUE_SIZE = 100;
+
+    private Sender sender;
+    private int flushInterval = DEFAULT_FLUSH_INTERVAL_MS;
+    private int maxQueueSize = DEFAULT_MAX_QUEUE_SIZE;
+    private Metrics metrics;
+    private boolean withLogging;
+
+    @Override
+    public Builder withLoggingReporter() {
+      this.withLogging = true;
+      return this;
+    }
+
+    public Builder withFlushInterval(int flushInterval) {
+      this.flushInterval = flushInterval;
+      return this;
+    }
+
+    public Builder withMaxQueueSize(int maxQueueSize) {
+      this.maxQueueSize = maxQueueSize;
+      return this;
+    }
+
+    public Builder withMetrics(Metrics metrics) {
+      this.metrics = metrics;
+      return this;
+    }
+
+    public Builder withSender(Sender sender) {
+      this.sender = sender;
+      return this;
+    }
+
+    @Override
+    public Reporter build() {
+      if (sender == null) {
+        sender = new UdpSender(UdpSender.DEFAULT_AGENT_UDP_HOST, UdpSender.DEFAULT_AGENT_UDP_COMPACT_PORT, 0);
+      }
+      if (metrics == null) {
+        metrics = new Metrics(new StatsFactoryImpl(new NullStatsReporter()));
+      }
+      Reporter reporter = new RemoteReporter(sender, flushInterval, maxQueueSize, metrics);
+      if (withLogging) {
+        reporter = new CompositeReporter(new LoggingReporter(), reporter);
```

Comment:
This is one reason why the separation of configuration exists. I think it breaks encapsulation if RemoteReporter suddenly becomes aware of things like logging reporter.

Reply:
It can be removed from here, I think it's better to add it directly in `TracerBuilder`